<?xml version="1.0"?>
<robot name="sad_robot" xmlns:xacro="http://www.ros.org/wiki/xacro">

<xacro:property name="hardware_interface" value="hardware_interface/PositionJointInterface"/>

    <link name="world"/>
    <joint name="base_fixed_joint" type="fixed" disable_fixed_joint_lumping="true">
        <origin rpy="0 0 0" xyz="0 0 0"/>
        <parent link="world"/>
        <child link="base_link"/>
    </joint>

    <link name="base_link" selfCollide="false">
        <inertial>
            <mass value="1.0"/>
            <inertia ixx="0.01" ixy="0" ixz="0" iyy="0.01" iyz="0" izz="0.01" />
        </inertial>  
        <collision>
            <origin xyz="0 0 0.151"/>
            <geometry>
                <box size = "0.25 0.25 0.302"></box> 
            </geometry>
        </collision>
        <visual>
            <origin xyz="0 0 0.151"/>
            <geometry>
                <box size = "0.25 0.25 0.302"></box>        
            </geometry>
        </visual>
    </link>
    <link name="link1" selfCollide="false">
        <inertial>
            <mass value="1.0"/>
            <inertia ixx="0.01" ixy="0" ixz="0" iyy="0.01" iyz="0" izz="0.01" />
        </inertial>  
        <collision>
            <geometry>
                <cylinder radius="0.05" length="0.2"/> 
            </geometry>
        </collision>
        <visual>
            <geometry>
                <cylinder radius="0.05" length="0.2"/>    
            </geometry>
        </visual>
    </link>
    <link name="link2" selfCollide="false">
        <inertial>
            <mass value="1.0"/>
            <inertia ixx="0.01" ixy="0" ixz="0" iyy="0.01" iyz="0" izz="0.01" />
        </inertial>  
        <collision>
            <geometry>
                <cylinder radius="0.039" length="0.2"/> 
            </geometry>
        </collision>
        <visual>
            <geometry>
                <cylinder radius="0.039" length="0.2"/> 
            </geometry>
        </visual>
    </link>
    <link name="link3" selfCollide="false">
        <inertial>
            <mass value="1.0"/>
            <inertia ixx="0.01" ixy="0" ixz="0" iyy="0.01" iyz="0" izz="0.01" />
        </inertial>  
        <collision>
            <geometry>
                <cylinder radius="0.1" length="0.2"/> 
            </geometry>
        </collision>
        <visual>
            <geometry>
                <cylinder radius="0.1" length="0.2"/> 
            </geometry>
        </visual>
    </link>
    <link name="link4" selfCollide="false">
        <inertial>
            <mass value="1.0"/>
            <inertia ixx="0.01" ixy="0" ixz="0" iyy="0.01" iyz="0" izz="0.01" />
        </inertial>  
        <collision>
            <geometry>
                <cylinder radius="0.02025" length="0.2"/> 
            </geometry>
        </collision>
        <visual>
            <geometry>
                <cylinder radius="0.02025" length="0.2"/> 
            </geometry>
        </visual>
    </link>
    <link name="link5" selfCollide="false">
        <inertial>
            <mass value="1.0"/>
            <inertia ixx="0.01" ixy="0" ixz="0" iyy="0.01" iyz="0" izz="0.01" />
        </inertial>  
        <collision>
            <geometry>
                <cylinder radius="0.1" length="0.02"/> 
            </geometry>
        </collision>
        <visual>
            <geometry>
                <cylinder radius="0.1" length="0.02"/> 
            </geometry>
        </visual>
    </link>
    <link name="link6" selfCollide="false">
        <inertial>
            <mass value="1.0"/>
            <inertia ixx="0.01" ixy="0" ixz="0" iyy="0.01" iyz="0" izz="0.01" />
        </inertial>  
        <collision>
            <geometry>
                <cylinder radius="0.01275" length="0.2"/> 
            </geometry>
        </collision>
        <visual>
            <geometry>
                <cylinder radius="0.01275" length="0.2"/> 
            </geometry>
        </visual>
    </link>
    <link name="link7" selfCollide="false">
        <inertial>
            <mass value="1.0"/>
            <inertia ixx="0.01" ixy="0" ixz="0" iyy="0.01" iyz="0" izz="0.1" />
        </inertial>  
        <collision>
            <geometry>
                <cylinder radius="0.1" length="0.02"/> 
            </geometry>
        </collision>
        <visual>
            <geometry>
                <cylinder radius="0.1" length="0.02"/> 
            </geometry>
        </visual>
    </link>
    <link name="link8" selfCollide="false">
        <inertial>
            <mass value="1.0"/>
            <inertia ixx="0.01" ixy="0" ixz="0" iyy="0.01" iyz="0" izz="0.01" />
        </inertial>  
        <collision>
            <geometry>
                <cylinder radius="0.01" length="0.14"/> 
            </geometry>
        </collision>
        <visual>
            <geometry>
                <cylinder radius="0.01" length="0.14"/> 
            </geometry>
        </visual>
    </link>

    <joint name="joint1" type="revolute">
        <parent link="base_link"/>
        <child link="link1"/>
        <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.402"/>
        <axis xyz="0 0 1" />
        <limit effort="1000.0" velocity="0.1" lower="-1.57" upper="1.57" />
        <dynamics damping="100" friction="50" />
    </joint>
    <joint name="joint2" type="revolute">
        <parent link="link1"/>
        <child link="link2"/>
        <origin rpy="-1.5707963267948966 0.0 0.0" xyz="0.0 0.0 0.139"/>
        <axis xyz="0 0 1" />
        <limit effort="1000.0" velocity="0.1" lower="-1.57" upper="1.57" />
        <dynamics damping="100" friction="50" />
    </joint>
    <joint name="joint3" type="revolute">
        <parent link="link2"/>
        <child link="link3"/>
        <origin rpy="1.5707963267948966 0.0 0.0" xyz="0.0 0.0 0.1"/>
        <axis xyz="0 0 1" />
        <limit effort="1000.0" velocity="0.1" lower="-1.57" upper="1.57" />
        <dynamics damping="100" friction="50" />
    </joint>
    <joint name="joint4" type="revolute">
        <parent link="link3"/>
        <child link="link4"/>
        <origin rpy="1.5707963267948966 0.0 0.0" xyz="0.0 0.0 0.02025"/>
        <axis xyz="0 0 1" />
        <limit effort="1000.0" velocity="0.1" lower="-1.57" upper="1.57" />
        <dynamics damping="100" friction="50" />
    </joint>
    <joint name="joint5" type="revolute">
        <parent link="link4"/>
        <child link="link5"/>
        <origin rpy="-1.5707963267948966 0.0 0.0" xyz="0.0 0.0 0.01"/>
        <axis xyz="0 0 1" />
        <limit effort="1000.0" velocity="0.1" lower="-1.57" upper="1.57" />
        <dynamics damping="100" friction="50" />
    </joint>
    <joint name="joint6" type="revolute">
        <parent link="link5"/>
        <child link="link6"/>
        <origin rpy="1.5707963267948966 0.0 0.0" xyz="0.0 0.0 0.01275"/>
        <axis xyz="0 0 1" />
        <limit effort="1000.0" velocity="0.1" lower="-1.57" upper="1.57" />
        <dynamics damping="100" friction="50" />
    </joint>
    <joint name="joint7" type="revolute">
        <parent link="link6"/>
        <child link="link7"/>
        <origin rpy="1.5707963267948966 0.0 0.0" xyz="0.0 0.0 0.01"/>
        <axis xyz="0 0 1" />
        <limit effort="1000.0" velocity="0.1" lower="-1.57" upper="1.57" />
        <dynamics damping="100" friction="50" />
    </joint>
    <joint name="joint8" type="revolute">
        <parent link="link7"/>
        <child link="link8"/>
        <origin rpy="1.5707963267948966 0.0 0.0" xyz="0.0 0.0 0.01"/>
        <axis xyz="0 0 1" />
        <limit effort="1000.0" velocity="0.1" lower="-1.57" upper="1.57" />
        <dynamics damping="100" friction="50" />
    </joint>
</robot>